Switch UI build system to pnpm (#605)
tomrtk/fish-code: src/ui/README.md

By using bundlers like `esbuild` and `postcss`. We are able to abstract some
of the burden of keeping dependencies up to date. Including the packages in
`package.json` we only need to update versions and rerun `pnpm run build:dev`

## How To Run

To be able to run the frontend, some step needs to be taken beforehand. It
depends on whether it runs in development or production.

### Compile changes to CSS/JavaScript

> Using the command below, but are compiled at once. View `pnpm run` for other options.

Reviewer: ```suggestion
Use the command below. View `pnpm run` for other options.
```
